model-optimizer/extensions/front/tf/IteratorGetNextCut.py

```diff
@@ -0,0 +1,45 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from collections import defaultdict
+
+from mo.front.extractor import add_input_ops
+from mo.graph.graph import Graph
+from mo.middle.passes.convert_data_type import SUPPORTED_DATA_TYPES, np_data_type_to_precision
+from mo.utils.error import Error
+from mo.front.common.replacement import FrontReplacementPattern
+
+
+class IteratorGetNextCut(FrontReplacementPattern):
+    """
+    Cuts OneShotIterator -> IteratorGetNext pattern
+    in order to enable Out Of the Box (OOB) usage.
+    Pass is run only if user didn't specify any inputs names and shapes.
+    """
+    enabled = True
+    graph_condition = [lambda graph: graph.graph['cmd_params'].input is None]
+
+    def run_before(self):
+        from extensions.front.output_cut import OutputCut
+        from extensions.front.input_cut import InputCut
+        return [OutputCut, InputCut]
+
+    def run_after(self):
+        return []
+
+    def find_and_replace_pattern(self, graph: Graph):
+        iter_get_next_shapes = defaultdict(list)
+        for iter_get_next in graph.get_op_nodes(op='IteratorGetNext'):
+            iter_get_next_name = iter_get_next.soft_get('name', iter_get_next.id)
+            for port in iter_get_next.out_nodes():
```

Comment:
please investigate why you have to use old API

Reply:
Problem with `out_ports()` was because of missing `out_ports_count` attribute. Usually this attribute is defined in operation definition in `mo/ops/op_name.py`. 

But since for `IteratorGetNext` we do not have operation but only an extractor i had to specify this attribute in the extractor
https://github.com/openvinotoolkit/openvino/pull/8040/files#diff-2c1e7af2ebbfb229767b8514884f6818a65132e16286a24e4eb37efca04bcc35